Added Traffic RPC.

melvinlu opened the pull request:
Added traffic RPC that utilizes Bing's traffic API. Included is an example & custom blocks. You will need a Bing traffic API key.

There is not much functionality to this RPC. Currently it just returns an array populated with coordinates of traffic incidents in a general area.

brollb commented:
Traffic type should probably be decoded on the server. That is, `type` should be "accident", "construction", etc, not 1, 2, etc:
![traffictype](https://cloud.githubusercontent.com/assets/4982789/17232401/6f49908c-54ed-11e6-98da-d977b39fe735.png)

Also, for severity, I think we should either return the severity for each accident or (probably the better choice) just remove it altogether and return them all. It isn't really clear what the scale is and it might be more cumbersome to explain it here (on the block).
![trafficseverity](https://cloud.githubusercontent.com/assets/4982789/17232397/661215e8-54ed-11e6-8a68-b306183f1ea0.png)

melvinlu commented:
Thanks. I went ahead and just removed the severity parameter (turns out I never actually implemented it with this commit).

brollb commented:
I just realized that the traffic messages don't stop when the map changes and clicking the map resets the map and starts the messages... I would probably make clicking only start the traffic messages and have the traffic messages stop and clear on map updates

melvinlu commented:
I'm not sure if  I understood you correctly, can you take a look at the change? I had the messages stop and clear on map updates. However, there is a limit on how "wide" the zoom can be for the API so I just force the map to be at zoom 5 whenever the messages come in.

brollb commented:
also, what happens if the area is too large? does it return nothing or just an incomplete set?

melvinlu commented:
How's that? (Changed `trigger` to be a reporter block that will return an error if the area is too large, and added the four navigation blocks) 

I also went ahead and added back in the zoom out block since the `trigger` block will now return an error.

brollb commented on src/server/rpc/libs/RPC:
```diff
@@ -8,3 +8,4 @@ weather.xml	Weather API
 pubsub.xml	PubSub
 kvstore.xml	Key-Value Store
 battleship.xml	Battleship
+Traffic.xml	Traffic
```
change `Traffic.xml` to `traffic.xml` - this will break on linux which is case sensitive

brollb commented:
One last fix and it should be good!

melvinlu commented:
Ah! My bad! There you go.

brollb commented:
awesome - thanks!
